dask_geomodeling/geometry/aggregate.py

```diff
@@ -397,23 +397,21 @@ def process(geom_data, raster_data, process_kwargs):
         # process in groups of disjoint subsets of the features
         agg = np.full((depth, len(features)), np.nan, dtype="f4")
         for select in bucketize(features.bounds.values):
-            agg_geometries_bucket = agg_geometries.iloc[select]
-            index = features.index[select]
-
             rasterize_result = utils.rasterize_geoseries(
-                agg_geometries_bucket,
+                agg_geometries.iloc[select],
                 process_kwargs["agg_bbox"],
                 agg_srs,
                 height,
                 width,
-                values=index,
+                values=np.asarray(select, dtype=np.int32),
```

Comment:
Why the int32? Is that to match the dtype of the index somehow?

Reply:
ratserize_geoseries will cast to int32 because GDAL only knows that one: https://github.com/nens/dask-geomodeling/blob/1ef8053529e827769de413bcbb74d1103868b2ef/dask_geomodeling/utils.py#L626